packages/@react-aria/grid/src/useGrid.ts

```diff
@@ -114,11 +130,10 @@ export function useGrid<T>(props: GridProps, state: GridState<T, GridCollection<
   let selection = state.selectionManager.rawSelection;
   let lastSelection = useRef(selection);
   useUpdateEffect(() => {
-    // Do not do this when using selectionBehavior = 'replace' to avoid selection announcements
-    // every time the user presses the arrow keys.
-    if (!state.selectionManager.isFocused || state.selectionManager.selectionBehavior === 'replace') {
-      lastSelection.current = selection;
-
+    // Do not announce every time the user presses the arrow keys.
+    let wasKeyboardNavigated = keyboardNavigated.current;
+    keyboardNavigated.current = false;
+    if (!state.selectionManager.isFocused || wasKeyboardNavigated) {
```

Comment:
Instead of checking the modality, maybe we could always announce if multiple items are selected (i.e. using shift), or the selection is cleared. In other words, announce except if there is only a single item is selected. The current solution doesn't announce at all when you are using the keyboard, and IMO it would be nice to know how many are selected.

Now that I think about it though, there should probably be some indication that the item you're navigating to is selected as well though. At least in VO, it only announces the row table, and doesn't read the `aria-selected` value. We should test other screen readers to ensure they also don't, but maybe we should just allow all announcements... 🤔  (If we do that, we'll need to modify the below logic so it actually reads the row contents and not just "1 item selected").

Reply:
this is a good point, lets talk tomorrow